lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -623,7 +606,7 @@ fn limit_str_to_len(s: &str, limit: usize) -> &str {
 /// `TagSplitter` can be cloned to create a new iterator with its own iteration state. The same underlying input byte
 /// slice is retained.
 #[derive(Clone)]
-struct TagSplitter<'a> {
+pub struct TagSplitter<'a> {
```

Comment:
It's fine for now, but more as a note to myself that I don't love this being exposed, and it'd be nice to find a clean way to avoid needing to expose this to support resolving the context.

It's _just_ weird/confusing enough of an `IntoIterator` impl (and in that same vein, weird how it's used) that my brain prefers it being hidden entirely.

Reply:
I agree it is a bit odd. I initially thought about returning something more like an array of tags, but this representation is nicely compact and sidesteps the issue of knowing ahead of time how many tags there might be to avoid heap allocation.

Reply:
Yeah, I have no problem with the implementation itself, just that I wish there was maybe a _little_ cleaner of an implementation that it could have given that it needs to be public now.